smithy-aws-traits/src/test/resources/software/amazon/smithy/aws/traits/errorfiles/http-checksum-header-conflicts.errors

```diff
@@ -1,9 +1,9 @@
 [SUPPRESSED] smithy.example#HeaderConflicts: This shape applies a trait that is unstable: aws.protocols#httpChecksum | UnstableTrait
 [SUPPRESSED] smithy.example#HeadersConflicts: This shape applies a trait that is unstable: aws.protocols#httpChecksum | UnstableTrait
 [SUPPRESSED] smithy.example#NoConflicts: This shape applies a trait that is unstable: aws.protocols#httpChecksum | UnstableTrait
-[WARNING] smithy.example#HeaderConflicts: The `httpHeader` binding of `x-amz-checksum-crc32` on `HeaderConflictError` starts with the prefix `x-amz-checksum-` used by the `httpChecksum` trait. | HttpChecksumTrait
-[WARNING] smithy.example#HeaderConflicts: The `httpHeader` binding of `x-amz-checksum-crc32` on `HeaderConflictsInput` starts with the prefix `x-amz-checksum-` used by the `httpChecksum` trait. | HttpChecksumTrait
-[WARNING] smithy.example#HeaderConflicts: The `httpHeader` binding of `x-amz-checksum-crc32` on `HeaderConflictsOutput` starts with the prefix `x-amz-checksum-` used by the `httpChecksum` trait. | HttpChecksumTrait
-[DANGER] smithy.example#HeadersConflicts: The `httpPrefixHeaders` binding of `HeadersConflictError` uses the prefix `x-amz-checksum-` that conflicts with the prefix `x-amz-checksum-` used by the `httpChecksum` trait. | HttpChecksumTrait
-[DANGER] smithy.example#HeadersConflicts: The `httpPrefixHeaders` binding of `HeadersConflictsInput` uses the prefix `x-amz-checksum-` that conflicts with the prefix `x-amz-checksum-` used by the `httpChecksum` trait. | HttpChecksumTrait
-[DANGER] smithy.example#HeadersConflicts: The `httpPrefixHeaders` binding of `HeadersConflictsOutput` uses the prefix `x-amz-checksum-` that conflicts with the prefix `x-amz-checksum-` used by the `httpChecksum` trait. | HttpChecksumTrait
+[WARNING] smithy.example#HeaderConflicts: The `httpHeader` binding of `x-amz-checksum-crc32` on `HeaderConflictError` starts with the prefix `x-amz-checksum-` used by the `httpChecksum` trait. | HttpChecksumTrait.httpHeader.HeaderConflictError.x-amz-checksum-crc32
```

Comment:
1. The pattern isn't consistent - use UpperCamel for all sections: `HttpChecksumTrait.HttpHeader.HeaderConflictError`
2. You probably don't need to say `HeaderConflictError`. Could this just be shortened to `HttpChecksumTrait.HeaderConflict`?
3. Do we need to sanitize the headers we put into the event ID at all, or does that not matter? I guess it doesn't matter since "." can just be ignored.

Reply:
(2) Sure yeah, appending to `HttpChecksumTrait.HeaderConflict` is fine -- `HttpChecksumTrait.HeaderConflict.x-amz-checksum-crc32`.

Reply:
Oh actually in this case "HeaderConflictError" is the name of the shape in the testcase:

operation HeadersConflicts {
    input: HeadersConflictsInput,
    output: HeadersConflictsOutput,
    errors: [HeadersConflictError,]
}

Reply:
Ah ok. There are probably other categories of errors this validator reports right? Like missing a member binding I'd guess. If that's the case, a category before the shape ID would be nice.

Reply:
Actually it's only prefix conflicts on the headers and prefix headers. It's redundant to have a category AND "httpHeader"/"httpPrefixHeaders", but I could replace the latter with categories if you prefer

Reply:
I like keeping the category in case we add future categories here. But I don't feel that strongly about it.